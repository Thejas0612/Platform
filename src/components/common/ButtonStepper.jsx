import { useDispatch, useSelector } from "react-redux";
import { ButtonInput } from "../dynamic-ui/uiComponentsConfig";
import "./buttonStepper.css";
import { changeAccordionStatus, schemaValidations } from "../../schema-service/schemaValidations";
import { changeActiveIndex } from "../../redux/reducers/initialBuDataSlice";

export default function ButtonStepper({ data, schema, updateValidations, buCode }) {
  const activeIndex = useSelector((state) => state.initialBuData?.activeIndex);
  const dispatch = useDispatch();

  const onNextChange = () => {
    const update_errors = schemaValidations(activeIndex, data, schema);
    if (update_errors?.isError) {
      updateValidations(update_errors?.update_schema, buCode);
      return;
    }
    dispatch(changeActiveIndex(activeIndex + 1));
    updateValidations(update_errors?.update_schema, buCode);
  };

  return (
    <div className="button_stepper_wrapper">
      {activeIndex > 0 ? (
        <ButtonInput
          label="Previous"
          showBackIcon={true}
          onClick={() => {
            dispatch(changeActiveIndex(activeIndex - 1));
            const navigation_changes = changeAccordionStatus(schema, activeIndex);
            updateValidations(navigation_changes, buCode);
          }}
        />
      ) : (
        <div></div>
      )}
      <ButtonInput
        label="Next"
        onClick={() => {
          onNextChange();
        }}
        size="sm"
      />
    </div>
  );
}
