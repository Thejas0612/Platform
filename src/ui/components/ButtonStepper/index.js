import ButtonStepper from './ButtonStepper.jsx';

export default ButtonStepper;