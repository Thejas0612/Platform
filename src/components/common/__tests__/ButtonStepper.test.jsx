import ButtonStepper from "../ButtonStepper";
import {render} from "@testing-library/react";
import {useDispatch, useSelector} from "react-redux";

jest.mock('react-redux')

const PROPS = {
    schema: {},
    updateValidations: ()=> {},
    data: [],
    buCode: 'test'
}

describe('<ButtonStepper/>', () => {
    beforeEach(() => {
        useDispatch.mockReturnValue(jest.fn())
        useSelector.mockReturnValue(0)
    })

    it('when index is 0, then only show next button', () => {
        const {container} = render(<ButtonStepper { ...PROPS}></ButtonStepper>)

        expect(container).toMatchSnapshot()
    })

})