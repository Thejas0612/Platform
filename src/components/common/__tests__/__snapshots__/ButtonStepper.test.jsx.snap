// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonStepper/> when index greater than 0, then only show next and previous button 1`] = `
<div>
  <div
    class="button_stepper_wrapper"
  >
    <a
      class="ddl-button-cta ddl-button-cta--primary  "
    >
      <i
        class="ddl-cta-icon ddl-cta-icon--prev-icon"
      />
      <span>
        Previous
      </span>
    </a>
    <a
      class="ddl-button-cta ddl-button-cta--primary  "
    >
      <span>
        Next
      </span>
      <i
        class="ddl-cta-icon ddl-cta-icon--internal-link"
      />
    </a>
  </div>
</div>
`;

exports[`<ButtonStepper/> when index is 0, then only show next button 1`] = `
<div>
  <div
    class="button_stepper_wrapper"
  >
    <div />
    <a
      class="ddl-button-cta ddl-button-cta--primary  "
    >
      <span>
        Next
      </span>
      <i
        class="ddl-cta-icon ddl-cta-icon--internal-link"
      />
    </a>
  </div>
</div>
`;
